This Notebook was created by Pablo Salazar

This notebook was used to create a Data Frame in Python from the data dowloaded from Wikipedia website "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"

Since, I was having issues to read the data from Wikipedia, I decided to download the data as a ".csv" file Then the data was pre-processed before uploading it in Python to create the Data Frame

Once the data was ready I used pandas to create the data frame in Python

In [1]:
import csv
import numpy as np
import pandas as pd

In [3]:
csv="C:\Post_Code_Data.csv"
df = pd.read_csv(csv)

In [4]:
df.head()

Postcode           Borough                     Neighbourhood
0      M3A        North York                         Parkwoods
1      M4A        North York                  Victoria Village
2      M5A  Downtown Toronto        Harbourfront, Regents Park
3      M6A        North York  Lawrence Heights, Lawrence Manor
4      M7A      Queen's Park                      Queen's Park

In [5]:
df.tail()

Postcode           Borough  \
98       M8X         Etobicoke   
99       M4Y  Downtown Toronto   
100      M7Y      East Toronto   
101      M8Y         Etobicoke   
102      M8Z         Etobicoke   

                                         Neighbourhood  
98       The Kingsway, Montgomery Road, Old Mill North  
99                                Church and Wellesley  
100  Business Reply Mail Processing Centre 969 Eastern  
101  Humber Bay, King's Mill Park, Kingsway Park So...  
102  Kingsway Park South West, Mimico NW, The Queen...

In [7]:
df.shape

(103, 3)

In the Data Frame I have a total of 103 lines, means 103 different post codes, and 3 columns

Then I dowloaded the "csv" file available with the Geospatial Coordinates

I did some pre-processing of the data and then used pandas to make the new data frame including Latitude and Longitude to the different Postal Code areas

In [3]:
import csv
import numpy as np
import pandas as pd
csv2="C:\Post_Code_Data_Lat_Long.csv"
df2 = pd.read_csv(csv2)

In [4]:
df2.head()

Postcode           Borough                     Neighbourhood  Latitude   \
0      M3A        North York                         Parkwoods  43.753259   
1      M4A        North York                  Victoria Village  43.725882   
2      M5A  Downtown Toronto        Harbourfront, Regents Park  43.654260   
3      M6A        North York  Lawrence Heights, Lawrence Manor  43.718518   
4      M7A      Queen's Park                      Queen's Park  43.662301   

   Longitude  
0 -79.329656  
1 -79.315572  
2 -79.360636  
3 -79.464763  
4 -79.389494

In [5]:
df2.tail()

Postcode           Borough  \
98       M8X         Etobicoke   
99       M4Y  Downtown Toronto   
100      M7Y      East Toronto   
101      M8Y         Etobicoke   
102      M8Z         Etobicoke   

                                         Neighbourhood  Latitude   Longitude  
98       The Kingsway, Montgomery Road, Old Mill North  43.653654 -79.506944  
99                                Church and Wellesley  43.665860 -79.383160  
100  Business Reply Mail Processing Centre 969 Eastern  43.662744 -79.321558  
101  Humber Bay, King's Mill Park, Kingsway Park So...  43.636258 -79.498509  
102  Kingsway Park South West, Mimico NW, The Queen...  43.628841 -79.520999

In [6]:
df2.shape

(103, 5)

Now I have the same 103 lines corresponding to different post codes and 2 additional columns (Latitude and Longitude)